src/EditorFeatures/CSharpTest/ReassignedVariable/CSharpReassignedVariableTests.cs

```diff
@@ -0,0 +1,853 @@
+}");
+        }
+
+        [Fact]
+        public async Task TestIndexerWithWriteInExprBody()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int this[int [|p|]] => [|p|]++;
+}");
+        }
+
+        [Fact]
+        public async Task TestIndexerWithWriteInGetter1()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int this[int [|p|]] { get => [|p|]++; }
+}");
+        }
+
+        [Fact]
+        public async Task TestIndexerWithWriteInGetter2()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int this[int [|p|]] { get { [|p|]++; } }
+}");
+        }
+
+        [Fact]
+        public async Task TestIndexerWithWriteInSetter1()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int this[int [|p|]] { set => [|p|]++; }
+}");
+        }
+
+        [Fact]
+        public async Task TestIndexerWithWriteInSetter2()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int this[int [|p|]] { set { [|p|]++; } }
+}");
+        }
+
+        [Fact]
+        public async Task TestPropertyWithAssignmentToValue1()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int Goo { set => [|value|] = [|value|] + 1; }
+}");
+        }
+
+        [Fact]
+        public async Task TestPropertyWithAssignmentToValue2()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int Goo { set { [|value|] = [|value|] + 1; } }
+}");
+        }
+
+        [Fact]
+        public async Task TestEventAddWithAssignmentToValue()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    event Action Goo { add { [|value|] = null; } remove { } }
+}");
+        }
+
+        [Fact]
+        public async Task TestEventRemoveWithAssignmentToValue()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    event Action Goo { add { } remove { [|value|] = null; } }
+}");
+        }
+
+        [Fact]
+        public async Task TestLambdaParameterWithoutReassignment()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    void M()
+    {
+        Action<int> a = x => Console.WriteLine(x);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestLambdaParameterWithReassignment()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    void M()
+    {
+        Action<int> a = [|x|] => Console.WriteLine([|x|]++);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestLambdaParameterWithReassignment2()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    void M()
+    {
+        Action<int> a = (int [|x|]) => Console.WriteLine([|x|]++);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestLocalWithoutInitializerWithoutReassignment()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    void M(bool b)
+    {
+        int p;
```

Comment:
Does this count as a reassignment of `p` ?

```cs
        int p;
        if (b)
            p = 1;

        p = 0;
        Console.WriteLine(p);
```

Reply:
yup!  there are codepaths with multiple writes.

Reply:
Could you add a test for this? I mean at `p=0`, `p` is not `DefinitelyAssignedOnEntry`, so from the code it would be ignored, right?

Reply:
added